### PR TITLE
Registration with email

### PR DESCRIPTION
**Pull request by yattias**
✅  Safe to release independently of corresponding FE portion

- Allows for registration via email
- Supports routes for changing password, and resetting password
- Email templates

### CODE REVIEW

File: src/user/custom_allauth.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from allauth.account.adapter import DefaultAccountAdapter, get_adapter
+from allauth.account.forms import ResetPasswordForm
+from allauth.account.utils import user_pk_to_url_str
+from dj_rest_auth.registration.serializers import RegisterSerializer
+from dj_rest_auth.serializers import PasswordResetSerializer
+
+from researchhub.settings import BASE_FRONTEND_URL
+
+
+class CustomAccountAdapter(DefaultAccountAdapter):
```
Comment (yattias) [inline src/user/custom_allauth.py:10]: Not sure if there is a more logical place to store these custom override classes

Comment (yattias) [reply src/user/custom_allauth.py:10]: The following are recommendations from author of DJ-Auth library, needed to override password reset, and change password mechanism

Comment (yattias) [reply src/user/custom_allauth.py:10]: Goal of which is to use a different frontend url than default

**Comment by le-sun** [comment]
Looks like the login tests are failing. Can you check those?

**Pull request merged**